frame/pallet-account/src/pallet/mod.rs

```diff
@@ -0,0 +1,228 @@
+                origin: new_origin,
+            });
+
+            Ok(())
+        }
+
+        /// Dispatch the given `call` from an account that the sender is authorised.
+        ///
+        /// The dispatch origin for this call must be _Signed_.
+        ///
+        /// Parameters:
+        /// - `origin_index`: Account origin index for using.
+        /// - `call`: The call to be made by the `derived` account.
+        #[pallet::weight({
+			let di = call.get_dispatch_info();
+			(T::WeightInfo::proxy_call()
+				.saturating_add(T::DbWeight::get().reads_writes(1, 1))
+				.saturating_add(di.weight),
+			di.class)
+		})]
+        #[pallet::call_index(1)]
+        pub fn proxy_call(
+            origin: OriginFor<T>,
+            #[pallet::compact] origin_index: u32,
+            call: Box<<T as Config>::RuntimeCall>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let custom_origin = AccountOrigin::<T>::get(who.clone())
+                .get(origin_index as usize)
+                .ok_or(Error::<T>::UnregisteredOrigin)?
+                .clone();
+
+            let e = if let Ok(id) = custom_origin.clone().try_into() {
+                // in case of native dispatch with system signed origin
+                call.dispatch(frame_system::RawOrigin::Signed(id).into())
+            } else {
+                // in other case dispatch with custom origin
+                call.dispatch(custom_origin.clone().into())
+            };
+
+            Self::deposit_event(Event::ProxyCall {
+                payer: who,
+                origin: custom_origin,
+                result: e.map(|_| ()).map_err(|e| e.error),
+            });
+
+            Ok(())
+        }
+
+        /// Send meta transaction: verify call signature and execute transaction.
+        ///
+        /// Signature should be made from following payload:
+        ///   * ChainMagic (set in pallet config)
+        ///   * Account Nonce (should be get form chain before call)
+        ///   * Runtime Call (well formatted and encoded runtime call)
+        ///
+        /// Fields above should be SCALE encoded and packed together. Signed and then
+        /// properly encoded into `MultiSignature` depend of selected signer key pair.
+        ///
+        /// The dispatch origin for this call must be _Signed_.
+        #[pallet::weight({
+			let di = call.get_dispatch_info();
+			(T::WeightInfo::meta_call()
+				.saturating_add(T::DbWeight::get().reads_writes(1, 1))
+				.saturating_add(di.weight),
+			di.class)
+		})]
+        #[pallet::call_index(2)]
+        pub fn meta_call(
+            origin: OriginFor<T>,
+            call: Box<<T as Config>::RuntimeCall>,
+            signer: T::AccountId,
+            signature: T::Signature,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let magic = T::ChainMagic::get();
+            let nonce = frame_system::Pallet::<T>::account_nonce(signer.clone());
```

Comment:
Signer usually doesn't have enough balance, and sometimes doesn't have it at all.
How do we hold nonce of accounts without Existential Deposit (ED)?

And because of ED, account info can be cleaned up, which means account nonce can be reset to 0, so I think it cannot ensure replay attack when that account is recreated later.

Being discussed here https://github.com/paritytech/substrate/issues/11988

Reply:
Agree with Shunsuke, `meta_call` should be left for some other PR in the future, potentially.

Reply:
Temporary removed